### PR TITLE
feature-3-separate-list

### LINKED ISSUE #3: Separate list into "In My Cart" and "Need to Buy"
Your shopping list should be broken into two sections. One for items that are already in your cart, and the other for items that you still need to buy. Again, try to make your components as modular and reusable as possible

### PR DESCRIPTION
**Pull request by allisonkopp**
closes #3

### CODE REVIEW

**Comment by Stommy123** [comment]
How do you determine which items are in the cart and which are in the wish list?

**Comment by Stommy123** [comment]
Hmm this is good for the next issue where you have to move the items back and forth between the carts. I had something simpler in mind.

**Comment by Stommy123** [comment]
Why not just give each an item another property, something like `inCart`. 
```
const { list } = this.state
const wishList = list.filter(item => !item.inCart)
const shoppingCart = list.filter(item => item.inCart)
return (
   <WishList items={wishList} />
   <ShoppingCart items={shoppingCart} />
)
```

**Comment by Stommy123** [comment]
All I wanted was a way for the two lists to differ. Not for you to actually implement adding/removing yet.

File: src/App.js [hunk 1]
```diff
@@ -1,14 +1,37 @@
...
+
+  addToCart = id => _ => {
+    const { list, shoppingCart } = this.state;
+    list.map(item => {
+      if (item.id === id) {
+        list.splice(item, 1);
+        return shoppingCart.push(item);
+      }
+    });
+    this.setState({ shoppingCart });
```
Comment (Stommy123) [inline src/App.js]: Also if you have an id, there's no need to map. You can splice at an index.

File: src/App.js [hunk 2]
```diff
@@ -1,14 +1,37 @@
...
+
+  addToWishlist = id => _ => {
+    const { list, shoppingCart } = this.state;
+    shoppingCart.map(item => {
+      if (item.id === id) {
+        shoppingCart.splice(item, 1);
+        return list.push(item);
+      }
+    });
+    this.setState({ list });
```
Comment (Stommy123) [inline src/App.js]: same comment as above.

File: src/components/Wishlist.jsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+const Wishlist = ({ list = [], addToCart }) => (
+  <>
+    <h1>Wishlist</h1>
+    <ul>
+      {list.map(item => (
+        <ListItem
+          name={item.name}
+          description={item.description}
+          toggleList={addToCart(item.id)}
+          itemLocation={'cart'}
```
Comment (Stommy123) [inline src/components/Wishlist.jsx]: You don't need jsx if the prop value is a string

File: src/App.js [hunk 4]
```diff
@@ -1,14 +1,23 @@
...
 import { data } from './data';
-import { ShoppingList } from './components';
+import { Wishlist, Cart } from './components';
 class App extends Component {
-  state = { list: data };
+  state = { list: data, shoppingCart: [] };
+
+  addToList = (id, status) => _ => {
+    const { list } = this.state;
+    if (list[id].id === id) list[id].inCart = status;
```
Comment (Stommy123) [inline src/App.js]: Mutating state directly is a big no. 
```
const { list } = this.state;
const newList = list.map(item => item.id === id ? { ...item, inCart: !item.inCart } : item)
this.setState({ list: newList })
```

File: src/App.js [hunk 5]
```diff
@@ -1,14 +1,23 @@
 import React, { Component } from 'react';
 import { data } from './data';
-import { ShoppingList } from './components';
+import { Wishlist, Cart } from './components';
 class App extends Component {
-  state = { list: data };
+  state = { list: data, shoppingCart: [] };
```
Comment (Stommy123) [inline src/App.js]: Lets remove this `shoppingCart` state since we're not using it anymore

File: src/App.js (hunk 4 above)
Comment (Stommy123) [reply src/App.js]: You also won't need that second argument with this method. The fewer arguments that a function relies on, the better.

File: src/App.js [hunk 6]
```diff
@@ -1,14 +1,23 @@
 import React, { Component } from 'react';
 import { data } from './data';
-import { ShoppingList } from './components';
+import { Wishlist, Cart } from './components';
 class App extends Component {
-  state = { list: data };
+  state = { list: data, newList: [] };
```
Comment (Stommy123) [inline src/App.js]: What is this `newList` state doing here! We removed an unused state to add another one??!

Comment (allisonkopp) [reply src/App.js]: Oh

**Pull request merged**